src/new/wallet.js

```diff
@@ -953,6 +881,10 @@ class HathorWallet extends EventEmitter {
         if (info.network.indexOf(this.conn.network) >= 0) {
           this.serverInfo = info;
           this.conn.start();
+          if (!this.storeSensitiveData) {
```

Comment:
I don't get it.

Wasn't it supposed to be:

```
if (this.storeSensitiveData) {
  this.pinCode = null;
  this.password = null;
}
```

To clear the `pinCode` and `password` from memory after a reconnect?

Reply:
The idea here is that we always need this info to create the wallet and encrypt data in storage. After the start succeeds we remove the pin/password from memory if I don't want to store sensitive data.

Reply:
Maybe we should optionally pass the pin and password in the `start(...)` method. So, if the user has set `this.pinCode` and `this.password`, they will be used. But, if they are empty, the user must pass both the pin and the password to the `start()` method.

What do you think? I feel that using the strategy we can remove the `this.storeSensitiveData` flag.

Reply:
Added as optional in `start` and `constructor`